cmd/eksctl-anywhere/cmd/downloadartifacts.go

```diff
@@ -0,0 +1,224 @@
+	Short:        "Download EKS Anywhere artifacts/manifests to a tarball on disk",
+	Long:         "This command is used to download the S3 artifacts from an EKS Anywhere bundle manifest and package them into a tarball",
+	PreRunE:      preRunDownloadArtifactsCmd,
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if err := downloadArtifacts(cmd.Context(), downloadArtifactsopts); err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+func downloadArtifacts(context context.Context, opts *downloadArtifactsOptions) error {
+	cliVersion := version.Get()
+	clusterSpec, err := cluster.NewSpec(opts.fileName, cliVersion)
+	if err != nil {
+		return err
+	}
+
+	release, err := clusterSpec.GetRelease(cliVersion)
+	if err != nil {
+		return err
+	}
+	bundlesManifestUrl := release.BundleManifestUrl
+
+	specManifests := []string{
+		bundlesManifestUrl,
+		clusterSpec.GetReleaseManifestUrl(),
+	}
+	for _, manifestURI := range specManifests {
+		if opts.dryRun {
+			logger.Info(fmt.Sprintf("Found artifact: %s\n", manifestURI))
+			continue
+		}
+		if err = downloadArtifact("", opts.downloadDir, manifestURI); err != nil {
+			return fmt.Errorf("error downloading artifact: %v", err)
+		}
+	}
+
+	bundle := clusterSpec.VersionsBundle
+	for component, manifestList := range bundle.Manifests() {
+		for _, manifest := range manifestList {
+			if opts.dryRun {
+				logger.Info(fmt.Sprintf("Found artifact: %s\n", manifest.URI))
+				continue
+			}
+			if err = downloadArtifact(component, opts.downloadDir, manifest.URI); err != nil {
+				return fmt.Errorf("error downloading artifact for component %s: %v", component, err)
+			}
+		}
+	}
+
+	if !opts.dryRun {
+		if err = createTarball(opts.downloadDir); err != nil {
+			return err
+		}
+
+		if !opts.retainDir {
+			if err = os.RemoveAll(opts.downloadDir); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func preRunDownloadArtifactsCmd(cmd *cobra.Command, args []string) error {
+	cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+		if err := viper.BindPFlag(flag.Name, flag); err != nil {
+			log.Fatalf("Error initializing flags: %v", err)
+		}
+	})
+	return nil
+}
+
+func downloadArtifact(component, downloadDir, artifactUri string) error {
+	logger.V(3).Info(fmt.Sprintf("Downloading artifact: %s", artifactUri))
+
+	fileURL, err := url.Parse(artifactUri)
+	if err != nil {
+		return err
+	}
+	path := fileURL.Path
+	segments := strings.Split(path, "/")
+	fileName := segments[len(segments)-1]
+
+	var filePath string
+	if component != "" {
+		filePath = filepath.Join(downloadDir, component, fileName)
+	} else {
+		filePath = filepath.Join(downloadDir, fileName)
+	}
+	if err := os.MkdirAll(filepath.Dir(filePath), 0o755); err != nil {
+		return err
+	}
+
+	logger.V(3).Info(fmt.Sprintf("Creating local artifact file: %s", filePath))
+	artifactFile, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer artifactFile.Close()
+
+	resp, err := http.Get(artifactUri)
```

Comment:
Should probably also support embedded manifest files

Reply:
Probably using the [methods](https://github.com/aws/eks-anywhere/blob/main/pkg/cluster/spec.go#L353) defined in spec.go